tests/network/vmi_networking.go

```diff
@@ -729,28 +732,31 @@ var _ = SIGDescribe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com]
 				table.Entry("with custom CIDR [IPv4]", []v1.Port{}, "10.10.10.0/24"),
 			)
 
-			table.DescribeTable("IPv6", func(ports []v1.Port) {
+			table.FDescribeTable("IPv6", func(ports []v1.Port, networkCIDR string) {
 				libnet.SkipWhenNotDualStackCluster(virtClient)
 				var serverVMI *v1.VirtualMachineInstance
 				var clientVMI *v1.VirtualMachineInstance
 
-				clientVMI, err = fedoraMasqueradeVMI([]v1.Port{})
+				clientVMI, err = fedoraMasqueradeVMI([]v1.Port{}, networkCIDR)
 				Expect(err).ToNot(HaveOccurred())
 				clientVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(clientVMI)
 				Expect(err).ToNot(HaveOccurred())
 				clientVMI = tests.WaitUntilVMIReady(clientVMI, console.LoginToFedora)
 
-				Expect(configureIpv6(clientVMI)).To(Succeed(), "failed to configure ipv6 on client vmi")
+				if networkCIDR == "" {
+					networkCIDR = api.DefaultVMIpv6CIDR
+				}
```

Comment:
IIUC, you're doing this to test the default value. Good.

I don't understand why you would change the value of an input parameter though. I'd have something like

```golang
expectedNetworkCIDR := networkCIDR
if networkCIDR == "" {
    expectedNetworkCIDR = api.DefaultVMIpv6CIDR
}
```
and compare against that new variable. It is more explicit and readable, does not change the value of an input parameter (something which I consider a bad practice), and cost a little bit of memory.

Will not make a fuss over this though.

Reply:
It is actually worse, I should not have used it here.
Fixed it now by passing it to where it is really needed. Still I update the input arg to a default, it feels natural to me, it is just a way to implement defaults in Go.

Reply:
I don't think so. Mutating an input argument is, imo, a bad idea, and bad practice. 

But hey, if the language cared about that, they'd have not allowed for you to mutate variables that are not declared w/ var. 

I'd just prefer to have immutable constructs everywhere; as I said, your call, not making a fuss over this.

Reply:
It is one of the ways to implement a "default" and I have not seen anyone reasoning why this is not allowed.
In C/C++ you have the `const` keyword, in others you can have a default in the signature. Here we can do this.

If you think that it is bad practice, I would love to know the arguments for it. You can take this case as an example if you like.

Reply:
> It is one of the ways to implement a "default" and I have not seen anyone reasoning why this is not allowed.
In C/C++ you have the const keyword, in others you can have a default in the signature. Here we can do this.

I think we're actually saying the same :) 

Above I wrote: "if the language cared about that, they'd have not allowed for you to mutate variables that are not declared w/ var."

> If you think that it is bad practice, I would love to know the arguments for it. You can take this case as an example if you like.

If you never mutate a variable, you will know at all times what it's value is. You can move code around more easily, since no matter where you put it, it'll have the same thing in it. Plenty more arguments about immutable variables / data structure can be found online.

But, again, your call. You want to mutate an input argument, I won't stand in your way. I am just pointing out that you could achieve the same thing in a safer way. Up to you.

Reply:
I agree that mutating an argument with different values in the same scope is problematic.
In this case, I think it is normalized in the begining and not in the middle, also not used for two different things (like the original code which was wrong).
In my opinion, with your reasoning in mind, the rule should be that a value of an argument should be stable from the first use. Until the first use, it can be adjusted, like with the default pattern.

Reply:
> I agree that mutating an argument with different values in the same scope is problematic.
> In this case, I think it is normalized in the begining and not in the middle, also not used for two different things (like the original code which was wrong).
> In my opinion, with your reasoning in mind, the rule should be that a value of an argument should be stable from the first use. Until the first use, it can be adjusted, like with the default pattern.

I keep thinking it should be normalized at creation time, not when it is used. E.g. it's first use may move. And the function behavior will change. 

It's creation time may move, but the compiler will blow up on your face if you use it before creating it, so it is safer.

And IMHO, 'defaulting' things has a time do be done: at creation time, nor when you don't find a value in it. I guess this is another facet of the discussion we're having :) 

But again: your call; if the language was meant to be used as I'm describing, it would have a feature set to enforce it.

Reply:
At this moment, I think it is fine as is. Having to name one variable/argument right is a hard, trying to name two that mean (almost) the same thing is even harder.

Reply:
Fair enough; thanks.